src/multiversioning.zig

```diff
@@ -1054,7 +1054,13 @@ pub const Multiversion = struct {
     }
 
     pub fn exec_latest(self: *Multiversion) !noreturn {
-        assert(self.stage == .ready);
+        // target_fd is only modified in target_update() which happens synchronously.
+        assert(self.stage != .target_update);
+
+        // Ensure that target_update() has been called at least once, and thus target_fd is
+        // populated by checking that target_header has been set.
+        assert(self.target_header != null);
+
         log.info("re-executing {s}...\n", .{self.exe_path});
```

Comment:
Is `\n`  intentional here?

Reply:
Yep - it gives a bit of visual separation in the logs from one version to the next.

Reply:
Might worth a comment. As it stands, it _will_ get cleanupd at some point by accident